src/cfg/cfg-traversal.h

```diff
@@ -101,6 +101,10 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
   // but their usage does not overlap in time, and this is more efficient.
   std::vector<std::vector<BasicBlock*>> processCatchStack;
 
+  // Variable to store the catch index within catch bodies. To be used in
+  // doStartCatch and doEndCatch.
+  Index catchIndex = 0;
```

Comment:
I may be wrong, but if there is a try-catch nested inside another catch, then they each need to operate on their own index, and so I think we need a stack?

If that's right, then I think we'd need an operation to push a 0 to the stack, to increment it, and to pop the stack?

Reply:
Ah right... I was very incorrect ;( Will fix.